synapse/logging/opentracing.py

```diff
@@ -964,7 +964,10 @@ def _decorator(func: Callable[P, R]) -> Callable[P, R]:
         if not opentracing:
             return func
 
-        return _custom_sync_async_decorator(func, _wrapping_logic)
+        # type-ignore: mypy seems to be confused by the ParamSpecs here.
```

Comment:
Is this a regression?

Reply:
I think so, but haven't dug very far into it. The error message is quoted in https://github.com/matrix-org/synapse/pull/15052/commits/9d1d8a64c85647a8c6a6edf2edc10537d936f1b0 , but I couldn't make much sense of it. Again it would be good to boil this down to a minimal test case.